lib/stub_ui/lib/src/ui/text.dart

```diff
@@ -1011,6 +1011,185 @@ enum BoxWidthStyle {
   max,
 }
 
+/// Data class that holds the metrics for a single line in the [Paragraph].
+class LineMetrics {
+
+  @pragma('vm:entry-point')
+  LineMetrics._(
+    this.startIndex,
+    this.endIndex,
+    this.endExcludingWhitespace,
+    this.endIncludingNewline,
+    this.hardBreak,
+    this.ascent,
+    this.descent,
+    this.unscaledAscent,
+    this.height,
+    this.width,
+    this.left,
+    this.baseline,
+    this.lineNumber,
+    List<RunMetrics> rMetrics,
+    List<int> rMetricsIndexes,
+  ) : runMetrics = Map() {
+    for (int i = 0; i < rMetrics.length; ++i) {
+      runMetrics[rMetricsIndexes[i]] = rMetrics[i];
+    }
+  }
+
+  /// The index in the text buffer the line begins.
+  final int startIndex;
+
+  /// The index in the text buffer the line ends.
+  final int endIndex;
+
+  /// The index in the text buffer the line ends ignoring whitespace.
+  ///
+  /// This value is provided as if any whitespace at the end were truncated.
+  final int endExcludingWhitespace;
+
+  /// The index in the text buffer the line ends including newline characters.
+  ///
+  /// This value is provided as if any trailing newline characters for the
+  /// current line were included. Newline characters in other lines are always
+  /// accounted for.
+  final int endIncludingNewline;
+
+  /// True if this line ends with a explicit line break (eg, '\n') or is the end
+  /// of the paragraph. False otherwise.
+  final bool hardBreak;
+
+  // The final computed ascent and descent for the line. This can be impacted by
```

Comment:
This comment will not show up in our published API docs. Is this on purpose? Feels like some of the information from this paragraph should be included with the corresponding properties?

Reply:
This was intentional, as the info here applies to multiple properties as a whole. Most of the information in here has been broken out into each relevant property, but not sure where to put the summary, as copying it multiple times seems a bit redundant.

For now, ill make sure all the points covered in the hidden docs are made clear in each property that is relevant.